File: ngmath/man/cssgrid/csc2s.m
.\"
.\"	$Id: csc2s.m,v 1.4 2008-07-27 03:35:35 haley Exp $
.\"
.TH CSC2S 3NCARG "May 2000" UNIX "NCAR GRAPHICS"
.SH NAME
CSC2S - convert from  Cartesian to  lat/lon coordinates
.SH SYNOPSIS
CALL CSC2S (N, X, Y, Z, RLAT, RLON) 
.SH DESCRIPTION
.IP N 12
(integer, input) The number of coordinates to convert.
.IP X,Y,Z 12
(real, input) The Cartesian coordinates of the input points. 
.IP RLAT 12
(real, output) The latitudes of the output coordinates, in degrees.
.IP RLON 12
(real, output) The longitudes of the output coordinates, in degrees.
.SH USAGE
CSC2S is called to find equivalent latitude and 
longitude coordinates to specified Cartesian coordinates
on the unit sphere. 
The coordinate (1.,0.,0.) is mapped to the
latitude/longitude coordinate (0.,0.). The latitude/longitude 
coordinates are returned in degrees. 
.SH ACCESS
To use CSC2S, load the NCAR Graphics library ngmath.
.SH SEE ALSO
css_overview,
cssgrid,
css2c.
.sp
Complete documentation for Cssgrid is available at URL
.br
http://ngwww.ucar.edu/ngdoc/ng/ngmath/cssgrid/csshome.html
.SH COPYRIGHT
Copyright (C) 2000
.br
University Corporation for Atmospheric Research
.br

The use of this Software is governed by a License Agreement.

File: ngmath/src/lib/gridpack/cssgrid/csc2s.f
C
C	$Id: csc2s.f,v 1.5 2008-07-27 03:10:06 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE CSC2S (N,X,Y,Z,RLAT,RLON)
      REAL X(N),Y(N),Z(N),RLAT(N),RLON(N)
C
C  This subroutine converts a point P from Cartesian coor-
C  dinates on a unit sphere to spherical coordinates.
C
C
C On input:
C
C       X,Y,Z = Cartesian coordinates of P.
C
C On output:
C
C       RLAT = Latitude of P, in degrees. - 90. .LE. RLAT .LE. 90.
C
C       PLON = Longitude of P, in degrees. -180. .LE. RLON .LE. 180.
C
C Modules required by CSC2S:  CSSCOORD
C
C Intrinsic functions called by CSC2S:  None
C
C***********************************************************
C
      PARAMETER (D2R=0.017453293, R2D=57.295778)
C
      DO 10 I=1,N
        CALL CSSCOORD(X(I),Y(I),Z(I),RLATD,RLOND,PRNM)
        RLAT(I) = R2D*RLATD
        RLON(I) = R2D*RLOND
   10 CONTINUE
C
      RETURN
      END
